tests/unit/scripts/test_config.py

```diff
@@ -14,31 +14,186 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-from typing import Any
+import os
+from pathlib import Path
+from typing import Any, Optional, List
+
+import pytest
 
-from cylc.flow.cfgspec.glbl_cfg import glbl_cfg
 from cylc.flow.scripts.config import get_config_file_hierarchy
+from cylc.flow.cfgspec.globalcfg import GlobalConfig
+
 
 Fixture = Any
+HOME: str = str(Path('~').expanduser())
+
+
+@pytest.fixture
+def conf_env(monkeypatch):
+    """Clear any env vars that effect which conf files get loaded.
+
+    Return a convenience function for setting environment variables.
+    """
+    # wipe any cached config
+    monkeypatch.setattr(
+        GlobalConfig,
+        '_DEFAULT',
+        None,
+    )
+
+    for envvar in ('CYLC_SITE_CONF_PATH', 'CYLC_CONF_PATH'):
+        if envvar in os.environ:
+            monkeypatch.delenv(envvar)
+
+    def _set_env(key, value):
+        if value:
+            monkeypatch.setenv(key, value)
+
+    return _set_env
+
+
+@pytest.fixture
+def dummy_version_hierarchy(monkeypatch):
+    """Set the config version hierarchy."""
+    monkeypatch.setattr(
+        'cylc.flow.cfgspec.globalcfg.GlobalConfig.VERSION_HIERARCHY',
+        ['', '1', '1.0']
+    )
+
+
+@pytest.fixture
+def capload(monkeypatch):
+    """Capture configuration load events.
+
+    This prevents actual file loading.
+
+    If the file name contains the string "invalid" it will not appear in the
+    results as if it diddn't exist on the filesystem.
+    """
+    files = []
 
+    def _capload(glblcfg, fname, _):
+        nonlocal files
+        if 'invalid' not in fname:
+            # if the file is called invalid skip it
+            # this is to replicate the behaviour of skipping files that
+            # don't exist
+            files.append(fname.replace(HOME, '~'))
 
-def test_get_config_file_hierarchy_global(monkeypatch: Fixture):
+    monkeypatch.setattr(
+        GlobalConfig,
+        '_load',
+        _capload
+    )
+    return files
+
+
+def test_get_config_file_hierarchy_global(
+    monkeypatch: Fixture,
+    conf_env: Fixture,
+    capload: Fixture,
+    dummy_version_hierarchy: Fixture
+):
```

Comment:
I didn't realise just by passing fixtures to test functions that they would be executed

Reply:
Up until the `return` or `yield` statement, for generators execution continues after the `yield` statement after the test scope exits.